src/PrestaShopBundle/Controller/Admin/Improve/Design/MailThemeController.php

```diff
@@ -242,6 +249,109 @@ public function previewThemeAction(Request $request, $themeName)
         ]);
     }
 
+    /**
+     * This action allows to send a test mail of a specific email template, however the Mail
+     * class used to send emails is not modular enough to allow sending templates on the fly.
+     * This would require either:
+     *  - a little modification of the Mail class to add an easy way to send a template content (rather than its name)
+     *  - a full refacto of the Mail class which wouldn't be coupled to static files any more
+     *
+     * These modifications will be performed in a future release so for now we can only send test emails
+     * with the current email theme using generated static files.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param string $theme
+     * @param string $layout
+     * @param string $locale
+     * @param string $module
+     *
+     * @return Response
+     *
+     * @throws InvalidArgumentException
+     */
+    public function sendTestMailAction($theme, $layout, $locale, $module = '')
+    {
+        if ($this->configuration->get('PS_MAIL_THEME') !== $theme) {
+            $this->addFlash(
+                'error',
+                $this->trans(
+                    'Cannot send test email for theme %theme% because it is not your current theme',
+                    'Admin.Notifications.Success',
```

Comment:
Are we sure of the localization here? I would have put it in _Admin.Notifications.Error_ instead, no?

Reply:
Nice catch!